### PR TITLE
Friend requests

### LINKED ISSUE #25: Friends system
Build out social features to allow users to send friend requests to one another

### PR DESCRIPTION
**Pull request by matthewdoles**
- Profile screen now has two list items: Quizes and Friends
- QuizResultsList screen now displays user's quiz results, navigated to from Profile screen
- Friends screen displays pending friend requests and current friends.
- Top list item on Friends screen navigates to UserSearch screen which is used to search for users by username
- When user navigates to another profile using UserSearch screen, user can send friend request, cancel sent friend request, and remove friend.
- If user for some reason doesn't use the pending request list on the Friends screen and navigates to the user's profile - it will also check to see if the user has a pending request from user to prevent duplicate friend requests
-  Update user global state on friend request accept and reject

### CODE REVIEW

File: consts/errorMessages.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export default {
+  friendRemove: 'Trouble removing friend, please try again.',
+  profileError: 'Trobule loading profile, please try again.',
```
Comment (quasney24) [inline consts/errorMessages.js]: lol not a big deal but you spelled trouble wrong

File: screens/Friends/index.js [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+const Friends = ({ navigation }) => {
+  const user = useSelector((state) => state.user.user);
+  const [friendRequests, setFriendRequests] = useState([]);
+  const [handlingFriendRequest, setHandlingFriendRequest] = useState(false);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    fetchRequests();
+  }, [dispatch]);
```
Comment (quasney24) [inline screens/Friends/index.js]: why is dispatch set as dep? wouldn't it be like friendRequests?

**Review by quasney24 — COMMENTED** [review]
Large PR lol, did more device testing than code review but it's a great start.

File: consts/errorMessages.js (hunk 1 above)
Comment (matthewdoles) [reply consts/errorMessages.js]: fixed

File: screens/Friends/index.js (hunk 2 above)
Comment (matthewdoles) [reply screens/Friends/index.js]: oh yeah not necessary, probably was using dispatch in useEffect at one point and didnt remove

**Pull request merged**